Products/Zuul/__init__.py

```diff
@@ -210,6 +210,32 @@ def filterUidsByPermission(dmd, permission, uids):
                 priority=messaging.WARNING)
     return validUids
 
+def checkAdministeredObjectPermission(uid, permission, context=None):
+    """
+    Return true if the current user has the specified permission as a admin role 
+    for the specified uid on the given context or the dmd; 
+    return none if user has no roles for requested item;
+    otherwise, return false.
+    
+    @type  uid: string
+    @param uid: The UID of the object to check for permissions.
+
+    @type  permission: string
+    @param permission: Permission we are looking up.
+
+    @type  context: DMD
+    @param context: you must pass in the dmd to this function to look up the objects.
+    """
+    context = context or get_dmd()
+    mobj = context.unrestrictedTraverse(uid)
+    if not mobj:
+        return None
+
+    for admin_role in mobj.adminRoles():
```

Comment:
This will allow *any* user to have permission if *any* user has the defined role on this object.
DANGEROUS, DO NOT MERGE THIS